bazel/PPROF.md

```diff
@@ -0,0 +1,131 @@
+# Memory consumption testing with `pprof`
+
+To use `pprof` to analyze performance and memory consumption in Envoy, you can
+use the built-in statically linked profiler, or dynamically link it in to a
+specific place yourself.
+
+# Linking
+
+## Static Linking
+
+Static linking is already available (because of a `HeapProfilerDump()` call
+inside
+[`Envoy::Profiler::Heap::forceLink()`](https://github.com/envoyproxy/envoy/blob/master/source/common/profiler/profiler.cc#L21-L26)).
+
+### Compiling a statically-linked Envoy
+
+Just compile Envoy as normal:
+
+    $ bazel build //source/exe:envoy-static
+
+### Running a statically-linked Envoy with `pprof`
+
+And run the binary with a `HEAPPROFILE` env var, like so:
+
+    $ HEAPPROFILE=/tmp/mybin.hprof bazel-bin/source/exe/envoy-static <args>
+
+## Dynamic Linking
+
+### Adding `tcmalloc_dep` to Envoy
+
+To add a `HeapProfiler` breakpoint yourself, add `tcmalloc` as a dependency
```

Comment:
nit: maybe add a one sentence explanation (or link) explaining what a heap profiler break point does?  Or, you could add a section (or link) at the top that explains the ways you can use the heap profiler.

Also, do heap profiler break points only work/apply in the dynamic linking setup?

Reply:
I believe the statically linked Envoy profiles everything by default.